### PR TITLE
scripts: Call bash through env

### PR DESCRIPTION
**Pull request by fabiankeil**
.. so the test scripts work on platforms like ElectroBSD and FreeBSD where bash usually isn't installed in /bin but in /usr/local/bin.

Fixes:

   [fk@test-vm ~/git/wolfssl]$ gmake check
   gmake -j3  check-recursive
   [...]
   FAIL: scripts/resume.test
   FAIL: scripts/tls13.test
   PASS: testsuite/testsuite.test
   ============================================================================
   Testsuite summary for wolfssl 5.5.1
   ============================================================================
   # TOTAL: 3
   # PASS:  1
   # SKIP:  0
   # XFAIL: 0
   # FAIL:  2
   # XPASS: 0
   # ERROR: 0

# Description

Please describe the scope of the fix or feature addition.

Fixes zd#

# Testing

How did you test?

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by dgarske** [comment]
OK to test. Contributor agreement on file

**Comment by douzzer** [comment]
Hi @fabiankeil 

There's no right answer here, but we would rather the scripts depend on `/bin/bash` being usable, than on `/usr/bin/env` being usable, and moreover, on `$PATH` having a value that doesn't result in surprises.

If we were to go down the `/usr/bin/env` route, there are at least 31 bash scripts in wolfssl that would need to be changed, and they would all become reliant on `env`, and all be subject to hard-to-troubleshoot misbehavior around surprising `$PATH` results.

The pedantically-correct solution here is for all scripts that use `bash` to have `.in` versions headed with `#!@bashpath@`, with `configure` detecting the right path for `bash` (per `$PATH`, subject to `--with-bash` override) and replacing `@bashpath@` to create the executable script.  As I noted above, by my cursory count there are 31 scripts that would need that treatment.
But there is a likely-fatal flaw with this solution: all of those scripts would become unusable outside autotools buildenvs/runtimes (of which we support a great number), which seems far too high a price to pay.

TL;DR: When and where needed, add a symlink at `/bin/bash` to your preferred `bash`.  I've always arranged for my own FreeBSD installations to have the same `bash` at `/bin/bash` as at `/usr/local/bin/bash`, and it's only ever been a problem solver for me.

**Comment by fabiankeil** [comment]
Understood. Thanks for the detailed response.
